### ISSUE TITLE
Document how to update *.o and *.a files

### ISSUE THREAD

**Issue by emersion** [andlabs/ui#149]
On Linux and macOS, seems that copying files from libui just works. On Windows though, I tried to do the same but I don't know if that is sufficient (so maybe it's what causing linker errors, #148).

I've also no idea where `static_windows_*.o` files come from.

**Comment by andlabs** [comment]
Yes, that much is still a bit of a black box. I'll work on it when I fix the other build errors people are having, as there are a few things wrong with the current setup.

**Comment by cryptix** [comment]
This would be awesome.

I tried fixing up this package to build under freebsd.
I got libui compiling as a standalone but couldn't figure out the build steps here.

**Comment by andlabs** [comment]
The general steps are:

- copy the out/libui.a as libui_freebsd_[arch].a
- copy the link_linux_[arch].go to link_freebsd_[arch].go as appropriate

The static .o files on Windows could possibly be converted to a `go generate` rule (at the cost of not being able to use a cross-compiler with the `go generate` step).

Were there any changes necessary to run on FreeBSD? I haven't tried it since I switched to cmake; I still need to set up the other BSDs to test them with.

**Comment by emersion** [comment]
> The static .o files on Windows could possibly be converted to a go generate rule (at the cost of not being able to use a cross-compiler with the go generate step).

How do you generate those currently?

**Comment by andlabs** [comment]
```
windres -O coff static.rc static.o
```

**Comment by cryptix** [comment]
> Were there any changes necessary to run on FreeBSD?

It complained a little about linking in the end (`-ldl` IIRC) but I didn't bother fiddling with include dir flags (it was late) and went with `-DBUILD_SHARED_LIBS=OFF`, which worked without any modifications.

**Comment by emersion** [comment]
We can also use `rsrc` instead of `windres`:
```
go get github.com/akavel/rsrc
rsrc -manifest test.manifest -o rsrc.syso
```

**Comment by andlabs** [comment]
I eventually decided that libui shouldn't have resources in static builds at all — they're just not supported in static libraries (presumably since they are global to the module you are linking the static library to). For package ui this means the manifest will make things a bit harder. I can provide a blank import for a generic manifest and then allow people to customize the manifest if they so choose, ensuring that the basics are still present. That rsrc tool seems interesting for the job...

Note to self: file libui issue about Windows icons automatically taking the first icon in the executable just like in Windows Explorer...

**Comment by andlabs** [comment]
libui now no longer has resources in static builds. Once Alpha 4 is up, package ui will also have this. I may need to make an optional underscore import package which will allow embedding a generic manifest file — those will still be needed — but (unless something interferes with that) other than that...
